engineapi/instance_manager.go

```diff
@@ -315,19 +316,27 @@ func getDataEngineFromInstanceProcess(p *imapi.Instance) longhorn.DataEngineType
 	return longhorn.DataEngineType(p.BackendStoreDriver)
 }
 
-func getTypeForInstance(instanceType longhorn.InstanceType, portCount int32) longhorn.InstanceType {
+func getTypeForInstance(instanceType longhorn.InstanceType, name string) longhorn.InstanceType {
 	if instanceType != longhorn.InstanceType("") {
 		return instanceType
 	}
 
-	if portCount == DefaultEnginePortCount {
+	// engine process name example: pvc-5a8ee916-5989-46c6-bafc-ddbf7c802499-e-0
+	nameSlices := strings.Split(name, "-")
+	processType := nameSlices[len(nameSlices)-2]
+
+	if processType == "e" {
 		return longhorn.InstanceTypeEngine
 	}
 	return longhorn.InstanceTypeReplica
 }
 
-func getTypeForProcess(portCount int32) longhorn.InstanceType {
-	if portCount == DefaultEnginePortCount {
+func getTypeForProcess(name string) longhorn.InstanceType {
```

Comment:
Can just call `getTypeForInstance(longhorn.InstanceNone, name)` in this function to prevent duplicated code.

Reply:
Fixed, thanks